sqlmesh/schedulers/airflow/operators/hwm_sensor.py

```diff
@@ -63,3 +63,33 @@ def _compute_target_high_water_mark(
         target_prev = to_datetime(target_snapshot.node.cron_floor(target_date))
         this_prev = to_datetime(self.this_snapshot.node.cron_floor(target_date))
         return min(target_prev, this_prev)
+
+
+class HighWaterMarkExternalSensor(BaseSensorOperator):
+    def __init__(
+        self,
+        snapshot: Snapshot,
+        external_table_sensor_factory: t.Callable[[t.Dict[str, t.Any]], BaseSensorOperator],
+        poke_interval: float = 60.0,
+        timeout: float = 7.0 * 24.0 * 60.0 * 60.0,  # 7 days
+        mode: str = "reschedule",
+        **kwargs: t.Any,
+    ):
+        super().__init__(
+            poke_interval=poke_interval,
+            timeout=timeout,
+            mode=mode,
+            **kwargs,
+        )
+        self.snapshot = snapshot
+        self.external_table_sensor_factory = external_table_sensor_factory
+
+    def poke(self, context: Context) -> t.Union[bool, PokeReturnValue]:
+        dag_run = context["dag_run"]
+        signals = self.snapshot.model.render_signals(
+            start=to_datetime(dag_run.data_interval_start),
+            end=to_datetime(dag_run.data_interval_end),
+            execution_time=now(),
```

Comment:
now or now_timestamp?

Reply:
All other arguments are datetime instances, so I figured I should be consistent.

Reply:
that's fine, just understand that it's minute floored

Reply:
oh, nice gotcha 😢